### ISSUE TITLE
Not compatible with `@html-eslint/eslint-plugin` when `language` is used instead of `languageOptions.parser` for parsing

### ISSUE THREAD

**Issue by andreww2012** [BenoitZugmeyer/eslint-plugin-html#312]
## Description

Hello! `@html-eslint/eslint-plugin` defines [an `html` language](https://github.com/yeonjuan/html-eslint/blob/b7f3d1d94a5583a1540fdb6e8a1a982cd6445dca/packages/eslint-plugin/lib/index.js#L17-L19) directly on the plugin, allowing to enable HTML parsing by setting

```ts
{
  language: '<@html-eslint/eslint-plugin prefix>/html'
}
```

instead of

```ts
import eslintParserHtml from '@html-eslint/parser';

{
  languageOptions: {
    parser: eslintParserHtml,
  },
}
```

However, if the method 1 is used instead of method 2 and `eslint-plugin-html` is added, HTML files linting stops completely.

## Setup configuration

- ESLint version: 9.39.2
- eslint-plugin-html version: 8.1.4
- NodeJS version: Not relevant
- Operating System name and version: Not relevant
- ESLint configuration (.eslintrc/eslint.config.js): in the repro

Reproduction: https://stackblitz.com/edit/vitejs-vite-c35nwdjv

1. Initially, method 1 is used and both plugins are added to the config. Run `pnpm test` to run ESLint on `index.html` and observe no found errors (expected: `1:1  error  '!DOCTYPE' is not in lowercase  @html-eslint/lowercase`)
2. Comment out `html: eslintPluginJsInHtml` line and run this script again. The problem will be reported now.
3. Uncomment the same line again and uncomment `languageOptions` section. The problem will continue to be correctly reported.

## Additional context

Very quickly looking at the source code of this plugin, I'm guessing this is expected, but this problem is quite hard to debug. I wonder if it's possible to somehow work this around or at least mention in the docs that explicitly specifying the parser is necessary? At the very least, do not suppress the reports from other plugins if the parser cannot be found. Thank you!

**Comment by andreww2012** [comment]
I managed to work this around with the following patch:

```patch
diff --git a/src/verifyWithFlatConfigPatch.js b/src/verifyWithFlatConfigPatch.js
index 8757a7a26fba9f1fe073d171a8ab397f67b673c6..ee7f9e3ade257efa155ef1ddce6ce70d4e0dfedf 100644
--- a/src/verifyWithFlatConfigPatch.js
+++ b/src/verifyWithFlatConfigPatch.js
@@ -146,9 +146,6 @@ function findExternalHtmlPluginName(config) {
 
 function verifyExternalHtmlPlugin(eslintModule, config, callOriginalVerify) {
   const htmlPluginName = findExternalHtmlPluginName(config)
-  if (!htmlPluginName) {
-    return [[], config]
-  }
 
   const rules = {}
   for (const name in config.rules) {
@@ -164,7 +161,8 @@ function verifyExternalHtmlPlugin(eslintModule, config, callOriginalVerify) {
         ...config.languageOptions,
         parser: eslintModule.require("espree"),
       },
-      rules,
+      ...(htmlPluginName && {rules}),
+      language: eslintModule.require('../languages/js'),
     }),
   ]
 }
```

If it is correct, I'm happy to submit a PR with the change!

**Comment by BenoitZugmeyer** [comment]
I updated the documentation regarding the compatibility with html-eslint https://github.com/BenoitZugmeyer/eslint-plugin-html#linting-html
